python/ray/train/predictor.py

```diff
@@ -144,22 +144,27 @@ def predict(self, data: DataBatchType, **kwargs) -> DataBatchType:
             DataBatchType: Prediction result. The return type will be the same as the
                 input type.
         """
-        data_df = convert_batch_type_to_pandas(data, self._cast_tensor_columns)
-
         if not hasattr(self, "_preprocessor"):
             raise NotImplementedError(
                 "Subclasses of Predictor must call Predictor.__init__(preprocessor)."
             )
 
-        if self._preprocessor:
-            data_df = self._preprocessor.transform_batch(data_df)
+        if not isinstance(data, (np.ndarray, dict)):
```

Comment:
We need to be checking if _predict_numpy/pandas/etc are defined for the given predictor (don't raise NotImplemented / are not equal to the base NotImplemented stub). Then, we should decide based on that and the data format the optimal one to call.

Btw, I noticed that we don't handle _predict_arrow here (the docs in this class seem to be wrong).